### PR TITLE
added test_scan_pe.py and with test files

### PR DESCRIPTION
**Pull request by infosec-intern**
**Describe the change**
Added a `tests` directory with tests for the ScanPe scanner

My version info parser was included in the tests, so if that does not get merged I can remove it from the tests script.

Since this is the only file being tested at the moment, I just went with the basic `unittests` module, but I'm reading up on other frameworks for testing across multiple files and test cases to see what else could work.

The `test_binary.c` file was compiled to create `test_binary.exe`, and it can be used to create an ELF file for testing using `make` pretty easily. I've never had to compile a MachO, nor do I have an Apple workstation to try, so it might require other dependencies to build on OSX.

**Describe testing procedures**
In the root of the strelka directory, simply run `python ./tests/test_scan_pe.py`. Directory paths should be built off the location of the test script, so it should work out of the box from this directory. For some reason I kept getting errors when running directly from the `tests` directory, but I wasn't able to figure out why

**Sample output**
```powershell
strelka> python .\tests\test_scan_pe.py
test_debug (__main__.ScanPeTests) ... ok
test_exports (__main__.ScanPeTests) ... skipped 'Need a DLL to test'
test_file_header (__main__.ScanPeTests) ... ok
test_imphash (__main__.ScanPeTests) ... ok
test_imports (__main__.ScanPeTests) ... ok
test_no_debug (__main__.ScanPeTests) ... ok
test_no_exports (__main__.ScanPeTests) ... ok
test_no_resources (__main__.ScanPeTests) ... ok
test_no_signature (__main__.ScanPeTests) ... ok
test_no_version_info (__main__.ScanPeTests) ... ok
test_no_warnings (__main__.ScanPeTests) ... ok
test_optional_header (__main__.ScanPeTests) ... ok
test_resources (__main__.ScanPeTests) ... ok
test_sections (__main__.ScanPeTests) ... ok
test_signature (__main__.ScanPeTests) ... skipped 'Need a PE file with an empty digital signature to test'
test_version_info (__main__.ScanPeTests) ... ok
test_warnings (__main__.ScanPeTests) ... skipped 'Need a PE file to test'

----------------------------------------------------------------------
Ran 17 tests in 1.410s

OK (skipped=3)
ScanPe test coverage: 82.4%
```

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of and tested my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jshlbrd** [comment]
@infosec-intern looks good, I still need to test it for myself, but one thought I had -- can we keep binary and other test files out of the repo and instead pull them in remotely when the test starts? If we add test files for each scanner type, then the size of the repo may get out of control (I've had this happen with other repositories ...)

**Comment by infosec-intern** [comment]
Makes sense, I can retrieve the file and throw it in a temp location during the testcase `setupClass` and delete it in the `tearDownClass` function

**Comment by infosec-intern** [comment]
New sample output. I reworked the test case to pull down Putty v0.70 at the start, and to delete it at the end. I need to add some other test files with perma-links to finish up the rest of the unit tests, but let me know what you think of this attempt.

```powershell
> python .\tests\test_scan_pe.py
test_debug (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_debug_none (__main__.ScanPeTests) ... ok
test_exports (__main__.ScanPeTests) ... skipped 'Need a DLL to test'
test_exports_none (__main__.ScanPeTests) ... ok
test_file_header (__main__.ScanPeTests) ... ok
test_imphash (__main__.ScanPeTests) ... ok
test_imports (__main__.ScanPeTests) ... ok
test_optional_header (__main__.ScanPeTests) ... ok
test_resources (__main__.ScanPeTests) ... ok
test_resources_none (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_sections (__main__.ScanPeTests) ... ok
test_signature (__main__.ScanPeTests) ... ok
test_signature_empty (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_signature_none (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_version_info (__main__.ScanPeTests) ... ok
test_version_info_none (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_warnings (__main__.ScanPeTests) ... skipped 'Need a PE file to test'
test_warnings_none (__main__.ScanPeTests) ... ok

----------------------------------------------------------------------
Ran 18 tests in 16.109s

OK (skipped=7)
ScanPe test coverage: 61.1%
```

**Comment by jshlbrd** [comment]
This looks great. If loading in remote files becomes a hassle, then we can setup a second repository to store files in. I still need to test it for myself but once I do, I'll merge it on or provide more feedback.

Once again, thanks for your effort on this, great work!

File: tests/test_scan_pe.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            "FileDescription": "SSH, Telnet and Rlogin client",
+            "InternalName": "PuTTY",
+            "OriginalFilename": "PuTTY",
+            "FileVersion": "Release 0.70",
+            "ProductVersion": "Release 0.70",
+            "LegalCopyright": "Copyright \u00a9 1997-2017 Simon Tatham."
+        }
+        version_info = {}
+        for entry in self.scanner.metadata["versionInfo"]:
+            version_info[entry["string-name"]] = entry["value"]
```
Comment (jshlbrd) [inline tests/test_scan_pe.py]: Finally got around to testing this for myself and I think the design is good for testing scanners (where the highest priority is ensuring the imported packages haven't altered outputted data). The only out of sync section I noticed when I ran the test is that the `string-name` field for the `versionInfo` dictionary should be `name` -- which is good, cause this means the unit test is doing its job! Feel free to make any additional changes you think are necessary and then I can merge when you're ready.

Comment (infosec-intern) [reply tests/test_scan_pe.py]: Good catch, I've fixed the test, so go ahead and merge whenever

**Pull request merged**